gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/documentation/PageRevisionCrudServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.crud_service.documentation;
+
+import io.gravitee.apim.core.documentation.crud_service.PageRevisionCrudService;
+import io.gravitee.apim.core.documentation.model.Page;
+import io.gravitee.apim.core.documentation.model.PageRevision;
+import io.gravitee.apim.core.exception.DomainException;
+import io.gravitee.apim.infra.adapter.PageAdapter;
+import io.gravitee.repository.exceptions.TechnicalException;
+import io.gravitee.repository.management.api.PageRevisionRepository;
+import io.gravitee.rest.api.service.exceptions.TechnicalManagementException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+public class PageRevisionCrudServiceImpl implements PageRevisionCrudService {
+
+    private final PageRevisionRepository pageRevisionRepository;
+    private static final Logger logger = LoggerFactory.getLogger(PageRevisionCrudServiceImpl.class);
+
+    public PageRevisionCrudServiceImpl(@Lazy PageRevisionRepository pageRevisionRepository) {
+        this.pageRevisionRepository = pageRevisionRepository;
+    }
+
+    @Override
+    public PageRevision create(Page page) {
+        var pageRevisionToCreate = PageAdapter.INSTANCE.toPageRevision(page);
+
+        this.computeHash(pageRevisionToCreate);
```

Comment:
I'm actually not sure it belongs in the ApiCreateDomainService because it's not really related to the domain at all, but of an attribute to be added when saving the PageRevision. 
It's not used anywhere else in the application. 
So for me, it would make sense for it to be only on the `infra` side of the application. It's tied closely with creating and updating PageRevision, so I would hesitate leaving it in the service, or moving it to another class on the `infra` side. WDYT?